redbot/cogs/trivia/trivia.py

```diff
@@ -671,9 +672,39 @@ async def _save_trivia_list(
                 return
 
         buffer = io.BytesIO(await attachment.read())
-        yaml.safe_load(buffer)
-        buffer.seek(0)
+        trivia_dict = yaml.safe_load(buffer)
+
+        schema = Schema(
+            {
+                Optional("AUTHOR"): str,
+                Optional("CONFIG"): {
+                    Optional("max_score"): int,
+                    Optional("timeout"): Or(int, float),
```

Comment:
Or(int, float) is required here instead of just float because schema will not interpret an integer value as a float. (e.g. the value 2 is valid but schema will throw an exception)